xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -811,3 +812,209 @@ bool CProcessorHD::IsBT2020Supported()
+    std::string name = DX::DXGIFormatToString(format);
+    if (name.find('R') != std::string::npos && name.find('G') != std::string::npos &&
+        name.find('B') != std::string::npos)
+      outputFormats.push_back(format);
+  }
+
+  // Color spaces directly supported by the swap chain - as a set for easy lookup
+  std::vector<DXGI_COLOR_SPACE_TYPE> bbcs = DX::DeviceResources::Get()->SwapChainColorSpaces();
+  std::set<DXGI_COLOR_SPACE_TYPE> backbufferColorSpaces(bbcs.begin(), bbcs.end());
+
+  std::string conversions;
+
+  // The input format cannot be worked around and is fixed.
+  // Loop over the lists of possible:
+  // - input color spaces
+  // - output formats
+  // - output color spaces
+  for (auto inputCS : ycbcrColorSpaces)
+  {
+    for (DXGI_FORMAT outputFormat : outputFormats)
+    {
+      for (DXGI_COLOR_SPACE_TYPE outputCS : rgbColorSpaces)
+      {
+        if (SUCCEEDED(m_pEnumerator1->CheckVideoProcessorFormatConversion(
+                inputFormat, inputCS, outputFormat, outputCS, &supported)) &&
+            supported == TRUE)
+        {
+          conversions.append("\n");
+          conversions.append(StringUtils::Format(
+              "{} {} / {}{} {:<{}} to {} {:<{}} / {}{} {:<{}}", "*",
+              DX::DXGIFormatToString(inputFormat),
+              (inputCS == heuristicsCS.inputColorSpace) ? "*" : " ",
+              (inputCS == inputNativeCS) ? "N" : " ", DX::DXGIColorSpaceTypeToString(inputCS), 32,
+              (outputFormat == heuristicsOutputFormat) ? "*" : " ",
+              DX::DXGIFormatToString(outputFormat), 26,
+              (outputCS == heuristicsCS.outputColorSpace) ? "*" : " ",
+              (backbufferColorSpaces.find(outputCS) != backbufferColorSpaces.end()) ? "bb" : "  ",
+              DX::DXGIColorSpaceTypeToString(outputCS), 32));
+        }
+      }
+    }
+  }
+
+  CLog::LogF(LOGDEBUG,
+             "supported conversions from format {}\n(*: values picked by "
+             "heuristics, N native input color space, bb supported as swap chain backbuffer){}",
+             DX::DXGIFormatToString(inputFormat), conversions);
+}
+
+std::vector<DXGI_FORMAT> CProcessorHD::ProcessorOutputFormats() const
+{
+  std::vector<DXGI_FORMAT> result;
+
+  UINT uiFlags;
+  for (int fmt = DXGI_FORMAT_UNKNOWN; fmt <= DXGI_FORMAT_V408; fmt++)
+  {
+    if (S_OK == m_pEnumerator->CheckVideoProcessorFormat((DXGI_FORMAT)fmt, &uiFlags) &&
+        uiFlags & D3D11_VIDEO_PROCESSOR_FORMAT_SUPPORT_OUTPUT)
+      result.push_back((DXGI_FORMAT)fmt);
+  }
+  return result;
+}
+
+DXGI_COLOR_SPACE_TYPE CProcessorHD::AvToDxgiColorSpace(const DXGIColorSpaceArgs& csArgs)
+{
+  // RGB
+  if (csArgs.color_space == AVCOL_SPC_RGB)
+  {
+    if (!csArgs.full_range)
+    {
+      if (csArgs.primaries == AVCOL_PRI_BT2020)
+      {
+        if (csArgs.color_transfer == AVCOL_TRC_SMPTEST2084)
+          return DXGI_COLOR_SPACE_RGB_STUDIO_G2084_NONE_P2020;
+
+        return DXGI_COLOR_SPACE_RGB_STUDIO_G22_NONE_P2020;
+      }
```

Comment:
@thexai in the same place, GetDXGIColorSpaceSource has an override to G22 when !supportHDR:

```
        if (csArgs.color_transfer == AVCOL_TRC_SMPTEST2084 && supportHDR)
          return DXGI_COLOR_SPACE_RGB_STUDIO_G2084_NONE_P2020;

        return DXGI_COLOR_SPACE_RGB_STUDIO_G22_NONE_P2020;

```
That was added with HDR support. But is there any way that the processor will ever see RGB input? Is the rest of the code actually ready for this?

Reply:
As commented the other day changes to `GetDXGIColorSpaceSource` seems not relevant for this PR aka "log conversions supported by the processor"

I would like to be able to separate these changes into another PR. I'm concerned about performance issues as these functions are called for each rendered frame 24 or 60 times / second and really have nothing to do with this PR (log conversions supported).

Reply:
yes, I'll force push. if you imported the branch you only have to pop the last two commits.

perf should be pretty close but I have other ideas to side step the issue completely.